include/arch/x86/asm-macros.h

```diff
@@ -45,15 +45,26 @@
 #define _ASM_IP    _ASM_REG(ip)
 #define _ASM_FLAGS _ASM_REG(flags)
 
+#define PT_PADDR(table, shift) (((.- (table)) / PTE_SIZE) << (shift))
+
 #ifdef __ASSEMBLY__
 
 /* clang-format off */
 .macro putc val
-  movw $0x3f8, %dx;
+  movw (com_ports), %dx;
```

Comment:
is there a way to show that you actually are using only the first one?

Reply:
I think it's enough to have *com_ports as uint16_t and movw takes only the first one. Did you have any other idea in mind?